packages/jellyfish-api-core/__tests__/category/icxorderbook/listHTLCs.test.ts

```diff
@@ -0,0 +1,409 @@
+import { ContainerAdapterClient } from '../../container_adapter_client'
+import { MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import {
+  ExtHTLC, HTLC, ICXClaimDFCHTLCInfo, ICXDFCHTLCInfo, ICXEXTHTLCInfo, ICXGenericResult, ICXListHTLCOptions,
+  ICXOfferInfo, ICXOrderInfo, ICXOffer, ICXOrder, ICXHTLCType, ICXHTLCStatus
+} from '../../../src/category/icxorderbook'
+import BigNumber from 'bignumber.js'
+import { setup, accountDFI, idDFI, checkDFISellOrderDetails, accountBTC, checkDFIBuyOfferDetails, checkDFCHTLCDetails, checkEXTHTLCDetails } from './common.test'
+
+describe('Should test ICXOrderBook.listHTLCs', () => {
+  const container = new MasterNodeRegTestContainer()
+  const client = new ContainerAdapterClient(container)
+
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForReady()
+    await container.waitForWalletCoinbaseMaturity()
+    await setup(container)
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  afterEach(async () => {
+    // cleanup code here
+  })
+
+  // common code for some tests until ICX offer
+  const setupUntilDFIBuyOffer = async (): Promise<string> => {
```

Comment:
Here the common setup is added to this function. If the setup code is longer than the intended test code in that test, sometimes it could be difficult to read/review the test. At the same time if too many references/common functions inside a test also could lead to some amount of unclarity.  So tried to find a middle ground.